packages/atomic/cypress/integration/result-list/result-components-v1/result-text.cypress.ts

```diff
@@ -0,0 +1,187 @@
+import {HighlightKeyword} from '@coveo/headless';
+import {generateComponentHTML} from '../../../fixtures/test-fixture';
+import {
+  interceptSearchResponse,
+  setUpPage,
+} from '../../../utils/setupComponent';
+import {
+  executeFirstSearch,
+  setFieldCaptions,
+} from '../../search-interface-utils.cypress';
+import {
+  generateResultList,
+  generateResultTemplate,
+  getFirstResult,
+} from '../result-list-v1-selectors';
+import {ResultTextSelectors} from './result-text-selectors';
+
+describe('Result Text Component', () => {
+  function setupResultTextPage(
+    props: Record<string, string | number>,
+    executeSearch = true
+  ) {
+    setUpPage(
+      generateResultList(
+        generateResultTemplate({
+          bottomMetadata: generateComponentHTML(
+            ResultTextSelectors.component,
+            props
+          ).outerHTML,
+        })
+      ),
+      executeSearch
+    );
+  }
+
+  function getFirstResultText() {
+    return getFirstResult().find(ResultTextSelectors.component);
+  }
+
+  describe('when not used inside a result template', () => {
+    beforeEach(() => {
+      setUpPage(generateComponentHTML(ResultTextSelectors.component).outerHTML);
+    });
+
+    it.skip('should remove the component from the DOM', () => {
+      cy.get(ResultTextSelectors.component).should('not.exist');
+    });
+
+    it('should log a console error', () => {
+      cy.get(ResultTextSelectors.component)
+        .find('atomic-component-error')
+        .should('exist');
+    });
+  });
+
+  describe('when the field does not exist for the result but the "default" prop is set', () => {
+    const defaultText = 'Test default';
+    beforeEach(() => {
+      setupResultTextPage({
+        default: defaultText,
+        field: 'thisfielddoesnotexist',
+      });
+    });
+
+    it('should render an "atomic-text" component with the default value', () => {
+      getFirstResultText()
+        .find('atomic-text')
+        .shadow()
+        .should('have.text', defaultText);
+    });
+  });
+
+  describe('when the field does not exist for the result and the "default" prop is not set', () => {
+    beforeEach(() => {
+      setupResultTextPage({
+        field: 'thisfielddoesnotexist',
+      });
+    });
+
+    it('should remove the component from the DOM', () => {
+      getFirstResultText().should('not.exist');
+    });
+  });
+
+  describe('when the field value is not a string', () => {
+    beforeEach(() => {
+      setupResultTextPage(
+        {
+          field: 420,
+        },
+        false
+      );
+      interceptSearchResponse((response) =>
+        response.results.forEach((result) => (result.raw['420'] = 'Abc'))
+      );
+      executeFirstSearch();
+    });
+
+    it.skip('should remove the component from the DOM', () => {
+      getFirstResultText().should('not.exist');
+    });
```

Comment:
Similar to the test for the number above, I'm not sure if this should show an error or remove the component.